packages/sdk/src/contracts/prebuilt-implementations/marketplace.ts

```diff
@@ -296,6 +296,39 @@ export class Marketplace implements UpdateableNetwork {
     return this.contractWrapper.readContract.timeBuffer();
   }
 
+  /**
+   * Get all the offers
+   *
+   * @remarks Fetch all the offers for a specified direct listing.
+   * @example
+   * ```javascript
+   * const offers = await marketplaceContract.getOffers(listingId);
+   * const firstOffer = offers[0];
+   * ```
+   *
+   * @param listingId - the id of the listing to fetch offers for 
+   */
+   public async getOffers(
+    listingId: BigNumberish,
+  ): Promise<(Offer)[]> {
+    // get all new offer events from this contract
+    const events = await this.events.getEvents("NewOffer");
+    // get only the events for this listing id
+    const listingEvents = events.filter((e) => e.data.listingId.eq(listingId));
```

Comment:
@joaquim-verges correct me if I'm wrong but I think this might be able to be combined in the `getEvents` via a filter call directly on-chain (meaning having to query less total listings?)

Reply:
there is a way to setup a filter passing the specific listing id, but you'd need to replicate some of the low level parsing we're doing. Don't think it's worth it right now. We can go with this approach first and then optimize it later.